File: engine/model/src/main/scala/net/server/channel/packet/cash/operation/SendGiftPacket.scala
package net.server.channel.packet.cash.operation

class SendGiftPacket(private var _action: Int, private var _birthday: Int, private var _sn: Int, private var _characterName: String, private var _message: String) extends BaseCashOperationPacket(_action = _action) {
  def birthday: Int = _birthday

  def sn: Int = _sn

  def characterName: String = _characterName

  def message: String = _message
}

File: engine/engine-model/src/main/java/net/server/channel/packet/cash/operation/SendGiftPacket.java
package net.server.channel.packet.cash.operation;

public class SendGiftPacket extends BaseCashOperationPacket {
   private final Integer birthday;

   private final Integer sn;

   private final String characterName;

   private final String message;

   public SendGiftPacket(Integer action, Integer birthday, Integer sn, String characterName, String message) {
      super(action);
      this.birthday = birthday;
      this.sn = sn;
      this.characterName = characterName;
      this.message = message;
   }

   public Integer birthday() {
      return birthday;
   }

   public Integer sn() {
      return sn;
   }

   public String characterName() {
      return characterName;
   }

   public String message() {
      return message;
   }
}
